### PR TITLE
[CE-165] Subscriber unable to edit venue/organizer

### PR DESCRIPTION
**Pull request by redscar**
### 🎫 Ticket

[CE-165]

### 🗒️ Description

When logged in as a subscriber the "Edit" link that displays would have an empty href value. This was due to there not being a return if the `$edit_link` was empty.

I added a bail statement if `$edit_link` is empty. As well as reversed some of the logic so it makes more sense.

As an administrator the logic still functions properly. 
image

As a subscriber the logic now does not output any link at all since `$edit_link` is empty.

image

[CE-165]: https://theeventscalendar.atlassian.net/browse/CE-165?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Linked_Posts/Chooser_Meta_Box.php [hunk 1]
```diff
@@ -186,23 +186,31 @@ protected function single_post_dropdown( $linked_post_id ) {
...
+		if ( ! $can_edit_others_posts ) {
 			return;
 		}
 
 		$edit_link = get_edit_post_link( $linked_post_id );
 
+		// Bail if $edit_link is empty.
+		if ( empty( $edit_link ) ) {
+			return;
+		}
```
Comment (Camwyn) [inline src/Tribe/Linked_Posts/Chooser_Meta_Box.php:212]: Might be more succinct to just do:
```php
// Bail if the user is unable to edit the type of post.
if (
    empty( $linked_post_pto->cap->edit_others_posts )
    || ! current_user_can( $linked_post_pto->cap->edit_others_posts )
) {
    return;
}
```

**Review by Camwyn — COMMENTED** [review]
Not necessary, but seems simpler and clearer to me?

File: src/Tribe/Linked_Posts/Chooser_Meta_Box.php (hunk 1 above)
Comment (Camwyn) [reply src/Tribe/Linked_Posts/Chooser_Meta_Box.php:212]: Sorry - meant for this to go on the `if ( ! $can_edit_others_posts ) {` line (203) 🤦🏼

Comment (redscar) [reply src/Tribe/Linked_Posts/Chooser_Meta_Box.php:210]: Good call on simplifying the logic. I just made the change :+1:

File: src/Tribe/Linked_Posts/Chooser_Meta_Box.php [hunk 2]
```diff
@@ -186,23 +186,29 @@ protected function single_post_dropdown( $linked_post_id ) {
...
-			empty( $linked_post_pto->cap->edit_others_posts )
-			|| ! current_user_can( $linked_post_pto->cap->edit_others_posts )
-		);
 
-		if ( is_admin() && $cannot_edit_others_posts ) {
+		// Bail if the user is unable to edit the type of post.
+		if (
+				empty( $linked_post_pto->cap->edit_others_posts )
+				|| ! current_user_can( $linked_post_pto->cap->edit_others_posts )
+		) {
```
Comment (Camwyn) [inline src/Tribe/Linked_Posts/Chooser_Meta_Box.php:201]:
```suggestion
		if (
			empty( $linked_post_pto->cap->edit_others_posts )
			|| ! current_user_can( $linked_post_pto->cap->edit_others_posts )
		) {
```
whitespace

**Review by Camwyn — CHANGES_REQUESTED** [review]
Small whitespace change - probably from my suggestion 🤦🏼

**Review by Camwyn — APPROVED** [review]

**Pull request merged**